openwisp_utils/base.py

```diff
@@ -3,23 +3,38 @@
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 from model_utils.fields import AutoCreatedField, AutoLastModifiedField
+from openwisp_utils.utils import get_random_key
+from openwisp_utils.validators import key_validator
 
 
-class TimeStampedEditableModel(models.Model):
+class UUIDModel(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class TimeStampedEditableModel(UUIDModel):
     """
     An abstract base class model that provides self-updating
     ``created`` and ``modified`` fields.
     """
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     created = AutoCreatedField(_('created'), editable=True)
     modified = AutoLastModifiedField(_('modified'), editable=True)
 
     class Meta:
         abstract = True
 
 
-class UUIDModel(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-
-    class Meta:
-        abstract = True
+class KeyField(models.CharField):
+    def __init__(self, unique: bool, db_index: bool, help_text: str, *args, **kwargs):
+        fields = {
+            'max_length': 64,
```

Comment:
I think it's better to make this parameter overridable

Reply:
@nemesisdesign Constructor accepts kwargs and then we update our fields dict with kwargs, so every parameter is overridable